CorsixTH/Lua/hospitals/player_hospital.lua

```diff
@@ -339,19 +339,20 @@ end
 --! Once a month the advisor may warn about long queues.
 --! Rooms requiring a doctor occasionally trigger the generic message
 function PlayerHospital:warnForLongQueues()
-  local queue_rooms, total_queue = {}, 0
+  local long_queue_rooms, active_rooms, total_queue = {}, 0, 0
   for _, room in pairs(self.world.rooms) do
-    if #room.door.queue then
+    if room.is_active and #room.door.queue then
       total_queue = total_queue + #room.door.queue
-    end
-    if #room.door.queue > 7 then
-      queue_rooms[#queue_rooms + 1] = room
+      active_rooms = active_rooms + 1
+      if #room.door.queue > 7 then
+        long_queue_rooms[#long_queue_rooms + 1] = room
+      end
     end
   end
-  if #queue_rooms == 0 or total_queue == 0 then return end
+  if #long_queue_rooms == 0 or total_queue == 0 then return end
 
-  local busy_threshold = 1.5 * total_queue / #self.world.rooms
-  local chosen_room = queue_rooms[math.random(1, #queue_rooms)]
+  local busy_threshold = 1.5 * total_queue / active_rooms
+  local chosen_room = long_queue_rooms[math.random(1, #long_queue_rooms)]
   if busy_threshold > #chosen_room.door.queue then return end
```

Comment:
There may not be rooms with a long queue, so `chosen_room` becomes `nil` (I hope).
In that case, `chosen_room.door.queue` will crash the program on dereferencing the `door` field from a `nil` value.

Reply:
If there aren't rooms with a long queue, then the function ends at line 352.

Reply:
good point :)